docs/modules/ROOT/pages/explanations/connectiondetails.adoc

```diff
@@ -0,0 +1,196 @@
+
+Let's assume our given helm chart writes a secret and a service.
+Both of them contain information relevant to the user to access the service.
+
+[source,golang]
+----
+r := &v1beta1.Release{
+		Spec: v1beta1.ReleaseSpec{
+			ResourceSpec: xpv1.ResourceSpec{
+				WriteConnectionSecretToReference: &xpv1.SecretReference{ <1>
+					Name: "myname",
+					Namespace: "secretnamespace",
+				},
+			},
+			ConnectionDetails: []v1beta1.ConnectionDetail{
+				{
+					ObjectReference: corev1.ObjectReference{
+						Kind:      "Secret",
+						Name:      "mysecret",
+						Namespace: "myns",
+						FieldPath: "data.my-password", <2>
+					},
+					ToConnectionSecretKey: "connectionKey",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
+						Kind:      "Service",
+						Name:      "myservice",
+						Namespace: "myns",
+						FieldPath: "spec.ports[0].port",
+					},
+				},
+			},
+		},
+	}
+----
+<1> The connection detail still needs to written somewhere
+<2> We don't have to bother with base64 encoding, provider-helm will handle this for us
+
+With this method it should be easier to handle most cases.
+However, provider-helm's connection detail management isn't perfect.
+It cannot handle any transformations on the secrets.
+So if anything needs to concatenated, split, or changed in any other way, then this might unfortunately not be the way to do it.
+
+== Unmanaged Connection Details
+
+There are cases where the connection details are in objects that aren't managed by the composition function or any of the providers.
+For example if the service is provisioned via an operator.
+Also, if the given methods to acquire the connection details aren't powerful enough, for example if some helm release connection details need some transformations.
+
+To still get access to any information needed, an observer object can be leveraged.
+This uses provider-kubernetes to deploy an `Object` with the `Observe` policy.
+These objects will only be read and never modified, making it perfect for this use-case.
+
+However this involes two steps, first to deploy the observer and then in the next reconcile to read it.
+
+. Deploy observer
+[source,golang]
+----
+secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mysecret",
+			Namespace: "myns",
+		},
+	}
+
+	return iof.Desired.PutIntoObserveOnlyObject(ctx, secret, comp.Name+"-secret-observer")
+----
+
+. Read observer
+[source,golang]
+----
+secret := &corev1.Secret{}
+err = iof.Observed.Get(ctx, secret, comp.Name+"-secret-observer")
+if err != nil {
+  if err == runtime.ErrNotFound { <1>
+    return runtime.NewNormal()
+  }
+	return runtime.NewFatalErr(ctx, "cannot get observed deletion job", err)
+}
+----
+<1> We need to ignore not found errors here, as the observer will not exist during the first reconcile
+
+Also, here for secrets the values will not be base64 encoded, so they can be used directly.
```

Comment:
I also expect some kind of documentation on how to add the gathered data to the composite here. Eg:
```
iof.Desired.PutCompositeConnectionDetail(ctx, v1alpha1.ExplicitConnectionDetail{
	Name:  "MY_PASSWORD",
	Value: string(secret.Data["my_password"])
})
```